plugin/al/plugin/AL_USDMayaTestPlugin/AL/usdmaya/commands/test_ProxyShapeSelect.cpp

```diff
@@ -288,25 +361,43 @@ TEST(ProxyShapeSelect, selectNode2)
     };
 
     auto compareNodes = [](const SdfPathVector& paths) {
-        MSelectionList sl;
-        MGlobal::getActiveSelectionList(sl);
-        EXPECT_EQ(sl.length(), paths.size());
-        for (uint32_t i = 0; i < sl.length(); ++i) {
-            MObject obj;
-            sl.getDependNode(i, obj);
-            MFnDependencyNode fn(obj);
-            MStatus           status;
-            MPlug             plug = fn.findPlug("primPath", &status);
-            EXPECT_EQ(MStatus(MS::kSuccess), status);
-            MString pathName = plug.asString();
-            bool    found = false;
-            for (uint32_t j = 0; j < paths.size(); ++j) {
-                if (pathName == paths[j].GetText()) {
-                    found = true;
-                    break;
+        if (!MayaUsdProxyShapePlugin::useVP2_NativeUSD_Rendering()) {
+            MSelectionList sl;
+            MGlobal::getActiveSelectionList(sl);
+            EXPECT_EQ(sl.length(), paths.size());
+
+            for (uint32_t i = 0; i < sl.length(); ++i) {
+                MObject obj;
+                sl.getDependNode(i, obj);
+                MFnDependencyNode fn(obj);
+                MStatus           status;
+                MPlug             plug = fn.findPlug("primPath", &status);
+                EXPECT_EQ(MStatus(MS::kSuccess), status);
+                MString pathName = plug.asString();
+                bool    found = false;
+                for (uint32_t j = 0; j < paths.size(); ++j) {
+                    if (pathName == paths[j].GetText()) {
+                        found = true;
+                        break;
+                    }
                 }
+                EXPECT_TRUE(found);
+            }
+        } else {
+            auto selection = Ufe::GlobalSelection::get();
+            for (auto it : *selection) {
+                bool found = false;
+                for (uint32_t j = 0; j < paths.size(); ++j) {
+                    auto str = it->path().string();
+                    auto index = str.find_first_of('/');
```

Comment:
Again, feels like you're only interested in the second segment of the path, so should use that directly, instead of converting to string and searching for the first '/'.